crates/compiler/parse/src/pattern.rs

```diff
@@ -347,24 +375,29 @@ fn loc_ident_pattern_help<'a>(
                 };
 
                 // Make sure `Foo Bar 1` is parsed as `Foo (Bar) 1`, and not `Foo (Bar 1)`
-                if can_have_arguments {
-                    let (_, loc_args, state) =
-                        loc_type_def_tag_pattern_args_help().parse(arena, state, min_indent)?;
-
-                    if loc_args.is_empty() {
-                        Ok((MadeProgress, loc_tag, state))
-                    } else {
-                        let region = Region::across_all(
-                            std::iter::once(&loc_ident.region)
-                                .chain(loc_args.iter().map(|loc_arg| &loc_arg.region)),
-                        );
-                        let value =
-                            Pattern::Apply(&*arena.alloc(loc_tag), loc_args.into_bump_slice());
-
-                        Ok((MadeProgress, Loc { region, value }, state))
-                    }
+                let (_, (args, style), state) = if can_have_arguments {
```

Comment:
Pedantically, I think we may only want `can_have_arguments` to influence whether it can have space-separated apply args. Not 100% sure on that. Perhaps worth some thought.